autotest/gcore/vrt_read.py

```diff
@@ -1666,9 +1669,17 @@ def test_vrt_protocol():
     ds = gdal.Open("vrt://data/tiff_with_subifds.tif?sd_name=1")
     assert ds.GetRasterBand(1).Checksum() == 35731
     with pytest.raises(Exception):
-        assert not gdal.Open("vrt://data/tiff_with_subifds.tif?sd=2&sd_name=1")
-        assert not gdal.Open("vrt://data/tiff_with_subifds.tif?sd=3")
-        assert not gdal.Open("vrt://data/tiff_with_subifds.tif?sd_name=sds")
```

Comment:
With multiple operations under the same `pytest.raises`, only one of them needs to fail.